mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -513,6 +513,7 @@ where
     // We don't need any explicit returns elsewhere as EOF should be an ERR.
     #[allow(clippy::cast_possible_truncation)]
     let all_zeroes: Vec<u8> = vec![0; *SERIALIZED_BLOCK_SIZE as usize];
+    let mut start = 0;
```

Comment:
Why do we need this change in logic? What was the bug exactly?

Also, tests?

Reply:
Bug is mainly using SeekFrom::Current instead of SeekFrom::Start. When we do something like this:

```rust
let mut buf = vec![];
file.read(&mut buf)?;
```
The seek pointer will move to the amount of bytes read. So if we did not find the block in the buffer then seek with current what happens will look like this:
Pretend following is true:
File size: 100
Buf size: 40
+------------------------+
| File data                         |
+------------------------+
 \        \  <---\----- SeekPtr is 40 now after read
  +-----+       |
   | buf   |        |
  +-----+       |
       \             |
        +-------+
         | block   | <- where block is really, say from 10-50 of the file
        +-------+

If we continue to use current then we will have read 0-40 then 40-80, thus missing the first block in the file. That is why we need to do something like SeekFrom::Start, so we instead do 0-40, 1-41, 2-42, etc...

Reply:
test added

Reply:
I am still a bit confused.. let's chat. 
But meanwhile, can you pls say what you said above to the PR as comments?